packages/labs/analyzer/src/lib/javascript/jsdoc.ts

```diff
@@ -109,34 +133,49 @@ export const parseNamedTypedJSDocInfo = (tag: ts.JSDocTag) => {
  * * @slot name description
  * * @slot name - description
  * * @slot name: description
+ * * @cssProp [--name=default]
+ * * @cssProp [--name=default] description
+ * * @cssProp [--name=default] - description
+ * * @cssProp [--name=default]: description
  */
 export const parseNamedJSDocInfo = (
   tag: ts.JSDocTag,
   requireDash = false
-): NamedDescribed | undefined => {
+): NamedDescribedDefault | undefined => {
   const comment = getJSDocTagComment(tag);
   if (comment == undefined) {
     return undefined;
   }
-  const nameDesc = comment.match(
-    requireDash ? parseNameDashDescRE : parseNameDescRE
-  );
-  if (nameDesc === null) {
-    throw new DiagnosticsError(
-      tag,
-      `Unexpected JSDoc format.${
-        parseNameDashDescRE
-          ? ` Tag must contain a whitespace-separated dash between the name and description, i.e. '@slot header - This is the description'`
-          : ''
-      }`
+  if (isOptionalJSDocNameRegex.test(comment)) {
```

Comment:
I think maybe a simpler way to structure this is to choose the regex based on `requireDash`, then follow the same code path regardless of default or not:

ie

```ts
const regex = requireDash ? parseNameDefaultDashDescRE : parseNameDefaultDescRE;
const result = comment.match(regex);
if (result === null) { /* error */ }
const {name, description, defaultValue} = nameDefaultDesc.groups!;
// ...
```

This requires making a regex that can handle both plain names and names inside balanced `[]`s. That'll be a little duplication in the regex, but will end up with fewer regex evaluations.

Reply:
PTAL, I think I have this for the non-dash case, but the dash case needs a little attention